app/src/main/java/org/wikipedia/dataclient/mwapi/EditorTaskCounts.kt

```diff
@@ -0,0 +1,150 @@
+package org.wikipedia.dataclient.mwapi
+
+import com.google.gson.JsonArray
+import com.google.gson.JsonElement
+import com.google.gson.annotations.SerializedName
+import org.wikipedia.json.GsonUtil
+import org.wikipedia.settings.Prefs
+import org.wikipedia.util.DateUtil
+import java.util.*
+
+class EditorTaskCounts {
+
+    @SerializedName("revert_counts") private val revertCounts: JsonElement? = null
+    @SerializedName("edit_streak") private val editStreak: JsonElement? = null
+
+    private val counts: JsonElement? = null
+
+    private val descriptionEditsPerLanguage: Map<String, Int>
+        private get() {
+            var editsPerLanguage: Map<String, Int>? = null
+            if (counts != null && counts !is JsonArray) {
+                editsPerLanguage =
+                    GsonUtil.getDefaultGson().fromJson(counts, Counts::class.java).appDescriptionEdits
+            }
+            return editsPerLanguage ?: emptyMap()
+        }
+    private val captionEditsPerLanguage: Map<String, Int>
+        private get() {
+            var editsPerLanguage: Map<String, Int>? = null
+            if (counts != null && counts !is JsonArray) {
+                editsPerLanguage =
+                    GsonUtil.getDefaultGson().fromJson(counts, Counts::class.java).appCaptionEdits
+            }
+            return editsPerLanguage ?: emptyMap()
+        }
+    val totalDepictsEdits: Int
+        get() {
+            var editsPerLanguage: Map<String, Int>? = null
+            if (counts != null && counts !is JsonArray) {
+                editsPerLanguage =
+                    GsonUtil.getDefaultGson().fromJson(counts, Counts::class.java).appDepictsEdits
+            }
+            return if (editsPerLanguage == null) 0 else (if (editsPerLanguage["*"] == null) 0 else editsPerLanguage["*"])!!
+        }
+    val totalEdits: Int
+        get() {
+            var totalEdits = 0
+            for (count in descriptionEditsPerLanguage.values) {
+                totalEdits += count
+            }
+            for (count in captionEditsPerLanguage.values) {
+                totalEdits += count
+            }
+            totalEdits += totalDepictsEdits
```

Comment:
Can be simplified as `descriptionEditsPerLanguage.values.sum() + captionEditsPerLanguage.values.sum() + totalDepictsEdits`